clients/fides-js/src/fides.ts

```diff
@@ -214,6 +220,20 @@ const init = async ({
     _Fides.geolocation = geolocation;
     _Fides.options = options;
     _Fides.initialized = true;
+    if (experience) {
+      // at this point, pre-fetched experience contains no user consent, so we populate with the Fides cookie
+      experience?.privacy_notices?.forEach((notice) => {
+        if (Object.hasOwn(cookie.consent, notice.notice_key)) {
+          // eslint-disable-next-line no-param-reassign
+          notice.current_preference = transformConsentToFidesUserPreference(
+            Boolean(cookie.consent[notice.notice_key]),
+            notice.consent_mechanism
+          );
+        } else {
```

Comment:
we do need this, since the FE components require that the `experience` contains all user preference info. But I agree, I'll hoist the logic out alongside `buildCookieConsentForExperiences`